toolchain/check/check.cpp

```diff
@@ -25,6 +30,20 @@ namespace Carbon::Check {
   auto Handle##Name(Context& context, Parse::Name##Id node_id) -> bool;
 #include "toolchain/parse/node_kind.def"
 
+// Handle suspending the definition of a function. This is used for inline
+// methods, which are processed out of the normal lexical order. This plus
+// HandleFunctionDefinitionResume carry out the same actions as
+// HandleFunctionDefinitionStart, except that the various context stacks are
+// cleared out in between.
+auto HandleFunctionDefinitionSuspend(Context& context,
+                                     Parse::FunctionDefinitionStartId node_id)
+    -> SuspendedFunction;
+
+// Handle resuming the definition of a function, after a previous suspension.
+auto HandleFunctionDefinitionResume(Context& context,
+                                    Parse::FunctionDefinitionStartId node_id,
+                                    SuspendedFunction sus_fn) -> void;
```

Comment:
I think it's very important that the definitions of these functions live next to `HandleFunctionDefinitionStart` and `HandleFunctionDefinition`, because they need to be maintained together and need to behave consistently.

How would you feel about adding a `handle_function.h`?

(As far as I'm aware, this is it in terms of suspending and resuming node actions. But who knows how we'll change the design in future.)

Reply:
handle_function.h seems okay, or handle.h both the main #define and this part.

Reply:
Ooh, I really like the `handle.h` idea. Thanks, done!